framework/src/modules/chain/synchronizer/fast_chain_switching_mechanism.js

```diff
@@ -325,18 +344,26 @@ class FastChainSwitchingMechanism extends BaseSynchronizer {
 				'Successfully switched chains. Node is now up to date',
 			);
 		} catch (err) {
-			this.logger.error({ err }, 'Error while processing blocks');
-			this.logger.debug(
-				{ height: highestCommonBlock.height },
-				'Deleting blocks after height',
-			);
-			await deleteBlocksAfterHeight(
-				this.processor,
-				this.blocks,
-				highestCommonBlock.height,
-			);
-			this.logger.debug('Restoring blocks from temporary table');
-			await restoreBlocks(this.blocks, this.processor);
+			if (err instanceof BlockProcessingError) {
```

Comment:
can you extract the error handling to private function?

Reply:
Done :)